frontend/ui/elements/font_settings.lua

```diff
@@ -61,10 +61,10 @@ function FontSettings:getAndroidPath()
     local user_path = android.getExternalStoragePath()
     if user_path == "Unknown" then
         -- unable to identify external storage. Use defaults
-        return system_path..";".."/sdcard/koreader/fonts"
+        return system_path..";".."/sdcard/fonts;/sdcard/koreader/fonts"
```

Comment:
Logically, shouldn't /sdcard/koreader/fonts/ comes first, even before system_path?
(e.g. if a user needs to tweak some fonts, even system fonts, to have them work better with koreader (interline space, weight), while keeping that original same font available for other android apps?)

(I hope the fonts we ship are checked for before all those in these directories. Is it the case?)

Reply:
The fonts we ship are stored in `/data/data/org.koreader.launcher/files`. These internal files cannot be modified by the user so we add a font dir under shared storage.

> Logically, shouldn't /sdcard/koreader/fonts/ comes first, even before system_path?
(e.g. if a user needs to tweak some fonts, even system fonts, to have them work better with koreader (interline space, weight), while keeping that original same font available for other android apps?)

Good point. I'm not a font expert and just add a few fonts myself that are not available as system fonts nor bundled. I'm not sure about the font loading mechanism and what happens if the same font is stored in a few different places.

Reply:
Well, we read the fonts in the directories in the order given here.
But at the end, we sort() them:
https://github.com/koreader/koreader/blob/0cbceca98a09e04783cb7fb176be8d72add501bf/frontend/fontlist.lua#L98-L105
And we feed that sorted set, in the sorted order, to crengine - which will ignore the duplicates (but crengine ignores the duplicate by font name, the internal name of the font - the filename doesn't matter).

So, it depends on the filename (if `table.sort()` keeps the original order for 2 identical elements - and if yes, good, the order of the directories would be preserved), and the internal name (to be or not considered a duplicate).

I guess a trick would be that if you want your font to be used first, wherever it is located: prefix its filename with "000"...

So, that order here is not 100% follproof, but it does still matter.